mifospay/src/main/java/org/mifos/mobilewallet/mifospay/standinginstruction/ui/SIDetailsActivity.kt

```diff
@@ -267,13 +261,51 @@ class SIDetailsActivity : BaseActivity(), StandingInstructionContract.SIDetailsV
     private fun showDiscardChangesDialog() {
         val dialogBox = DialogBox()
         dialogBox.setOnPositiveListener { dialog, which ->
+            fab.hide()
             dialog.dismiss()
-            finish()
+            editDetails(false);
```

Comment:
please remove `;`  xD, also why removing `finish()` and making effort to re-set the textviews when we simply need to exit the activity on clicking _accept_

Reply:
Oooh sorry for that I got with the flow of java I made this mistake several times during this, I think I left one here😄, I'll resolve it in few minutes.

Reply:
@EGOR-IND can you explain the reasoning behind the _re-setting_ of textviews before exiting and removing the `finish()` line, I might be missing something.

Reply:
Yeah, for sure,  the reason why i removed finish() from the positive button of dialog box is that it was directly exiting the activity, which was the whole purpose of this PR making the exiting experience more better, so when user accept discarding changes then they will first go to the read only version of activity and then they can exit. You can see what my changes are doing from the video😊.